### PR TITLE
[MU4] New score wizard. Template preview

### PR DESCRIPTION
**Pull request by RomanPudashkin**
Refactored TemplatePaintView to enable the use of input devices (like mouse and trackpad) on the template preview

### CODE REVIEW

File: src/notation/view/notationpaintview.h [hunk 1]
```diff
@@ -149,20 +155,19 @@ private slots:
...
 
     QColor m_backgroundColor;
     notation::INotationPtr m_notation;
     QTransform m_matrix;
-    NotationViewInputController* m_inputController = nullptr;
-    PlaybackCursor* m_playbackCursor = nullptr;
-    NoteInputCursor* m_noteInputCursor = nullptr;
+    QScopedPointer<NotationViewInputController> m_inputController;
+    QScopedPointer<PlaybackCursor> m_playbackCursor;
+    QScopedPointer<NoteInputCursor> m_noteInputCursor;
```
Comment (igorkorsukov) [inline src/notation/view/notationpaintview.h:166]: Maybe `std` is better? (for unification )

File: src/userscores/CMakeLists.txt [hunk 2]
```diff
@@ -51,6 +51,7 @@ set(MODULE_SRC
     )
 
 set(MODULE_LINK libmscore)
+set(MODULE_LINK notation)
```
Comment (igorkorsukov) [inline src/userscores/CMakeLists.txt:54]: This code removes `libmscore` from MODULE_LINK and adds `notation` to MODULE_LINK, there should probably be something else ... maybe need to remove `set(MODULE_LINK libmscore)`

File: src/notation/view/notationpaintview.cpp [hunk 3]
```diff
@@ -177,54 +191,45 @@ void NotationPaintView::onViewSizeChanged()
...
-    if (!interaction) {
-        return nullptr;
-    }
+INotationPlaybackPtr NotationPaintView::notationPlayback() const
+{
+    return currentNotation() ? currentNotation()->playback() : nullptr;
+}
 
-    return interaction->noteInput();
+INotationNoteInputPtr NotationPaintView::currentNotationNoteInput() const
```
Comment (igorkorsukov) [inline src/notation/view/notationpaintview.cpp:215]: maybe just `notationNoteInput()`?

File: src/notation/view/notationpaintview.cpp [hunk 4]
```diff
@@ -177,54 +191,45 @@ void NotationPaintView::onViewSizeChanged()
...
+{
+    return notationInteraction() ? notationInteraction()->noteInput() : nullptr;
 }
 
 INotationElementsPtr NotationPaintView::currentNotationElements() const
```
Comment (igorkorsukov) [inline src/notation/view/notationpaintview.cpp:220]: maybe just `notationElements()`?

File: src/notation/view/notationpaintview.cpp [hunk 5]
```diff
@@ -177,54 +191,45 @@ void NotationPaintView::onViewSizeChanged()
...
-    auto notation = currentNotation();
-    if (!notation) {
-        return nullptr;
-    }
-
-    return notation->elements();
+    return currentNotation() ? currentNotation()->elements() : nullptr;
 }
 
 INotationStylePtr NotationPaintView::currentNotationStyle() const
```
Comment (igorkorsukov) [inline src/notation/view/notationpaintview.cpp:225]: maybe just `notationStyle()`?

File: src/notation/view/notationpaintview.h [hunk 6]
```diff
@@ -90,23 +87,33 @@ class NotationPaintView : public QQuickPaintedItem, public IControlledView, publ
...
+    QRectF notationContentRect() const;
+
+protected slots:
+    virtual void onViewSizeChanged();
+
 private:
     INotationPtr currentNotation() const;
     INotationNoteInputPtr currentNotationNoteInput() const;
     INotationElementsPtr currentNotationElements() const;
     INotationStylePtr currentNotationStyle() const;
```
Comment (igorkorsukov) [inline src/notation/view/notationpaintview.h:113]: maybe better all names without `current`?

File: src/notation/view/notationpaintview.cpp [hunk 7]
```diff
@@ -421,15 +417,15 @@ void NotationPaintView::adjustCanvasPosition(const QRectF& logicRect)
     if (showRect.left() < viewRect.left()) {
         pos.setX(showRect.left() - border);
     } else if (showRect.left() > viewRect.right()) {
-        pos.setX(showRect.right() - width() / scale() + border);
+        pos.setX(showRect.right() - width() / currentScaling() + border);
```
Comment (Eism) [inline src/notation/view/notationpaintview.cpp:420]: please replace to  _scale

File: src/notation/view/notationpaintview.cpp [hunk 8]
```diff
@@ -421,15 +417,15 @@ void NotationPaintView::adjustCanvasPosition(const QRectF& logicRect)
...
     } else if (viewRect.width() >= showRect.width() && showRect.right() > viewRect.right()) {
         pos.setX(showRect.left() - border);
     }
 
     if (showRect.top() < viewRect.top() && showRect.bottom() < viewRect.bottom()) {
         pos.setY(showRect.top() - border);
     } else if (showRect.top() > viewRect.bottom()) {
-        pos.setY(showRect.bottom() - height() / scale() + border);
+        pos.setY(showRect.bottom() - height() / currentScaling() + border);
```
Comment (Eism) [inline src/notation/view/notationpaintview.cpp:428]: please replace to  _scale

Comment (RomanPudashkin) [reply src/notation/view/notationpaintview.cpp]: fixed

File: src/notation/view/notationpaintview.cpp (hunk 7 above)
Comment (RomanPudashkin) [reply src/notation/view/notationpaintview.cpp]: fixed

File: src/notation/view/notationpaintview.h (hunk 6 above)
Comment (RomanPudashkin) [reply src/notation/view/notationpaintview.h]: fixed

File: src/userscores/CMakeLists.txt (hunk 2 above)
Comment (RomanPudashkin) [reply src/userscores/CMakeLists.txt:53]: removed

File: src/notation/view/notationpaintview.h (hunk 1 above)
Comment (RomanPudashkin) [reply src/notation/view/notationpaintview.h]: replaced

**Review by Eism — APPROVED** [review]

**Pull request merged**